Packs/SymantecEDR/Integrations/SymantecEDR/SymantecEDR.py

```diff
@@ -0,0 +1,2367 @@
+            self.access_token = last_access_token
+            demisto.debug(f"Last login access token still active. Return token {last_access_token}")
+        else:
+            try:
+                response = self._http_request(
+                    method='POST',
+                    url_suffix='/atpapi/oauth2/tokens',
+                    auth=(self.client_key, self.secret_key),
+                    data={'grant_type': 'client_credentials'},
+                    resp_type='response'
+                )
+                response.raise_for_status()
+
+                new_access_token = response.json().get("access_token")
+                self.access_token = new_access_token
+                timestamp_string = int(time.time() * 1000)
+                demisto.debug(f"login: success, saving access token {new_access_token}\n,"
+                              f"Created Timestamp : {timestamp_string}")
+
+                if global_integration_context := demisto.getIntegrationContext():
+                    global_integration_context['access_token'] = new_access_token
+                    global_integration_context['created_timestamp_access_token'] = timestamp_string
+                    demisto.setIntegrationContext(global_integration_context)
+                else:
+                    demisto.setIntegrationContext({
+                        'access_token': new_access_token,
+                        'created_timestamp_access_token': timestamp_string
+                    })
+
+            except requests.exceptions.HTTPError as err:
+                status = response.status_code
+                if status in HTTP_ERRORS:
+                    raise DemistoException(f'{HTTP_ERRORS[status]}, '
+                                           f'Error from API: {response.json().get("error")},'
+                                           f'{response.json().get("message")}',
+                                           res=response)
+                else:
+                    # if it is unknown error - get the message from the error itself
+                    raise DemistoException(f'Failed to execute. Error: {str(err)}', res=response)
+
+    def query_request_api(self, method: str, url_suffix: str, params: dict[str, Any] = None,
+                          json_data: dict[str, Any] = None) -> Dict[str, Any]:
+        """
+        Call Symantec EDR On-prem POST and GET Request API
+        Args:
+            method (str): Request Method support POST and GET
+            url_suffix (str): API endpoint
+            params (dict): URL parameters to specify the query for GET.
+            json_data (dict): The dictionary to send in a request for POST.
+
+        Returns:
+            Return the raw api response from Symantec EDR on-premise API.
+        """
+        try:
+            response = self._http_request(
+                method=method.upper(),
+                url_suffix=url_suffix,
+                headers=self.headers,
```

Comment:
`headers` may raise DemistoException - this is not handled here. Is the exception necessary? let's discuss it.